draft-ietf-quic-transport.md

```diff
@@ -736,6 +733,18 @@ and will contain a CRYPTO frame with an offset matching the size of the CRYPTO
 frame sent in the first Initial packet.  Cryptographic handshake messages
 subsequent to the first do not need to fit within a single UDP datagram.
 
+### Handling of Fatal Initial Packets
+
+The contents of some Initial packets may, according to this specification, force
+connection termination. For example, they might contain forbidden frame types
+or a CONNECTION_CLOSE frame. As Initial packets are not protected, these could
+indicate injection attacks to terminate the connection.
+
+Endpoints MAY treat the receipt of such packets as a connection error, drop them
```

Comment:
The more I think about this, I think this is basically an intractable problem unless one is willing to keep copies of the QUIC handshake state machinery.

1) If you start processing a packet, and process one frame, then the next frame is invalid, you'd have to rewind/discard the state machinery.  Is that what you're recommending?
2) You have text that indicates you should process Initial packets, but not treat them as errors once you have received a Handshake packet, but similar to my first concern, I think that's fairly impractical unless you keep copies of the handshake.

To be clear, I sort of hate implicit ACKs of Initial, but we keep making decisions, such as that only Handshake packets count as address confirmation, that make anything sent in Initial largely useless.  And now I have https://github.com/quicwg/base-drafts/pull/1862 to fix the rest of the deadlock case, but that would be fixed if we didn't send ACKs in Initial and relied upon implicit ACKs instead.

Reply:
So why is there such resistance against implicit ACKs in handshake, when the solve a concrete problem, as opposed to a being a whacky optimization?

Reply:
I believe we've also said previously we don't care about man on the side attacks, but I think it's clear we do, but we don't want to do a full analysis of them, which leaves us in a weird place.

Reply:
There is nothing you can really do about injection attacks during the Initial phase of the handshake. This PR is more about the 3RTT (or, for lazy implementations, infinite) time that the initial keys remain lying around for use. This radically increases the time where a connection is vulnerable (especially if the server uses the client-generated CID, so that Initial keys are easy to generate).

In that context, I'm not sure how you would meaningfully process a late-arriving Initial Packet. Say the attacker sends the server Initial Packet #2 well after the handshake, with ACK, CRYPTO, and CONNECTION_CLOSE frames. Since Initial exchange is actually done, there are no outstanding packets to ACK. Since I've already started handshake CRYPTO exchange, it's too late to deliver any new CRYPTO data. So it is basically impossible to use frames to alter connection state except to bring down the connection by generating an error.

If your code sends off an ACK before processing the frames, that is a problem, but that seems like a bad thing to do for other reasons.